distribution/tools/keystore-cli/src/test/java/org/elasticsearch/common/settings/AddStringKeyStoreCommandTests.java

```diff
@@ -70,105 +80,92 @@ public void testMissingNoCreate() throws Exception {
+        terminal.addSecretInput(password);
         terminal.addSecretInput("and yet another secret value");
         execute("--force", "foo"); // force
-        assertSecureString("foo", "and yet another secret value");
+        assertSecureString("foo", "and yet another secret value", password);
     }
 
     public void testForceNonExistent() throws Exception {
-        createKeystore("");
+        String password = "keystorepassword";
+        createKeystore(password);
+        terminal.addSecretInput(password);
         terminal.addSecretInput("value");
         execute("--force", "foo"); // force
-        assertSecureString("foo", "value");
+        assertSecureString("foo", "value", password);
     }
 
     public void testPromptForValue() throws Exception {
-        KeyStoreWrapper.create().save(env.configFile(), new char[0]);
+        String password = "keystorepassword";
+        KeyStoreWrapper.create().save(env.configFile(), password.toCharArray());
+        terminal.addSecretInput(password);
         terminal.addSecretInput("secret value");
         execute("foo");
-        assertSecureString("foo", "secret value");
+        assertSecureString("foo", "secret value", password);
     }
 
     public void testStdinShort() throws Exception {
-        KeyStoreWrapper.create().save(env.configFile(), new char[0]);
+        String password = "keystorepassword";
+        KeyStoreWrapper.create().save(env.configFile(), password.toCharArray());
+        terminal.addSecretInput(password);
         setInput("secret value 1");
         execute("-x", "foo");
-        assertSecureString("foo", "secret value 1");
+        assertSecureString("foo", "secret value 1", password);
     }
 
     public void testStdinLong() throws Exception {
-        KeyStoreWrapper.create().save(env.configFile(), new char[0]);
+        String password = "keystorepassword";
+        KeyStoreWrapper.create().save(env.configFile(), password.toCharArray());
+        terminal.addSecretInput(password);
         setInput("secret value 2");
         execute("--stdin", "foo");
-        assertSecureString("foo", "secret value 2");
-    }
-
-    public void testStdinNoInput() throws Exception {
-        KeyStoreWrapper.create().save(env.configFile(), new char[0]);
-        setInput("");
-        execute("-x", "foo");
-        assertSecureString("foo", "");
-    }
-
-    public void testStdinInputWithLineBreaks() throws Exception {
-        KeyStoreWrapper.create().save(env.configFile(), new char[0]);
-        setInput("Typedthisandhitenter\n");
-        execute("-x", "foo");
-        assertSecureString("foo", "Typedthisandhitenter");
-    }
-
-    public void testStdinInputWithCarriageReturn() throws Exception {
-        KeyStoreWrapper.create().save(env.configFile(), new char[0]);
-        setInput("Typedthisandhitenter\r");
-        execute("-x", "foo");
-        assertSecureString("foo", "Typedthisandhitenter");
-    }
-
-    public void testAddUtf8String() throws Exception {
```

Comment:
What happened to these tests?

Reply:
These were removed in https://github.com/elastic/elasticsearch/pull/38498 but apparently by mistake, I don't remember removing them for a reason. I'll push them back in

Reply:
Good catch @pugnascotia ! Not only these were removed by mistake, but the removal was hiding a bug :pray: